fileinfo.go

```diff
@@ -43,6 +67,46 @@ func SetFileBasicInfo(f *os.File, bi *FileBasicInfo) error {
 	return nil
 }
 
+type rawFileStandardInfo struct {
+	AllocationSize, EndOfFile uint64
+	NumberOfLinks             uint32
+	DeletePending, Directory  uint32 /* BOOLEAN */
+}
+
+// FileStandardInfo contains extended information for the file.
+type FileStandardInfo struct {
+	AllocationSize, EndOfFile uint64
+	NumberOfLinks             uint32
+	DeletePending, Directory  bool
+}
+
+func (raw *rawFileStandardInfo) toPublic() *FileStandardInfo {
+	cooked := &FileStandardInfo{
+		AllocationSize: raw.AllocationSize,
+		EndOfFile:      raw.EndOfFile,
+		NumberOfLinks:  raw.NumberOfLinks,
+		DeletePending:  false,
+		Directory:      false,
+	}
+	if raw.DeletePending != 0 {
+		cooked.DeletePending = true
+	}
+	if raw.Directory != 0 {
+		cooked.Directory = true
+	}
+	return cooked
+}
+
+// GetFileStandardInfo retrieves ended information for the file.
+func GetFileStandardInfo(f *os.File) (*FileStandardInfo, error) {
+	si := &rawFileStandardInfo{}
```

Comment:
do we need to use an intermediate struct here? can't this be just `&FileStandardInfo{}`?

Reply:
I did this to get the `bool` in the public API to save lots of `!=0` checks for the users. I'm not sure if a `bool` in Go would match a `uint32 /* BOOLEAN */` safely when we're passing unsafe pointers into the C API.

Reply:
the public API makes sense. we did have a recent change, where we wrapped `BOOLEAN` with `bool`: https://github.com/microsoft/go-winio/blob/master/vhd/vhd.go#L54-L58.

Reply:
maybe @dcantah has more thoughts on this

Reply:
The OpenVirtualDisk options take in `BOOL`'s which is a typedef for int so I'm probably in the wrong here. For type BOOL for functions we'd been using just bool but for these we'd need the padding.

Reply:
Yeah, looking at `OpenVirtualDisk2` and how it flows into `_openVirtualDisk`, I suspect that if `procOpenVirtualDisk` took a "size" parameter for the parameters, it would have been failing because the struct was too small.

Reply:
So after all that, it turns out `BOOLEAN` _is_ 1 byte, not four. I don't know where I got `int32` from before, but the [protocol spec for driver developers](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-fscc/5afa7f66-619c-48f3-955f-68c4ece704ae) shows that it's two bytes for two booleans, followed by two bytes of padding.

Reply:
Possibly got `BOOL` and `BOOLEAN` mixed up as I tend to do regularly... haha. Although `BOOL` isn't unsigned so that parts a mystery

Reply:
ok... `BOOL` vs `BOOLEAN`